plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codetest/CodeTestException.kt

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.codetest
+
+import software.aws.toolkits.resources.message
+
+open class CodeTestException(
+    override val message: String?,
+    val statusCode: String? = "400",
+    val code: String? = "DefaultError",
+    val uiMessage: String? = message(
+        "testgen.error.generic_error_message"
+    ),
+) : RuntimeException()
+
+internal fun noFileOpenError(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.no_file_open"), "400", "ProjectZipError")
+
+internal fun fileTooLarge(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.file_too_large_telemetry"), "400", "ProjectZipError")
+
+internal fun cannotFindFile(errorMessage: String, filepath: String): Nothing =
+    error(message("codewhisperer.codescan.file_not_found", filepath, errorMessage))
+
+internal fun cannotFindValidFile(errorMessage: String): Nothing =
+    throw CodeTestException(errorMessage, "400", "ProjectZipError")
+
+internal fun cannotFindBuildArtifacts(errorMessage: String): Nothing =
+    throw CodeTestException(errorMessage, "400", "ProjectZipError")
+
+internal fun invalidSourceZipError(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.invalid_source_zip_telemetry"), "400", "InvalidSourceZipError")
```

Comment:
i am pretty sure none of these are actually error code 400, please use the actual error code from the service

Reply:
These are client side errors so there wont be any error codes from service

Reply:
if it is client side error then why would we have an http status code

Reply:
Its an ask from manager to see the statusCodes accordingly in the Kibana to differentiate the errors.